src/fides/api/ops/schemas/messaging/messaging.py

```diff
@@ -132,47 +153,76 @@ class Config:
         extra = Extra.forbid
 
 
-class EmailConfigRequest(BaseModel):
-    """Email Config Request Schema"""
+class MessagingServiceSecretsTwilioSMS(BaseModel):
+    """The secrets required to connect to twilio SMS."""
+
+    twilio_account_sid: str
+    twilio_auth_token: str
+    twilio_messaging_service_sid: Optional[str]
+    twilio_sender_phone_number: Optional[
+        str
+    ]  # Either the twilio_messaging_service_id *OR* the twilio_sender_phone_number should be supplied.
+
+    class Config:
+        """Restrict adding other fields through this schema."""
+
+        extra = Extra.forbid
+
+
+class MessagingServiceSecretsTwilioEmail(BaseModel):
+    """The secrets required to connect to twilio email."""
+
+    twilio_api_key: str
+
+    class Config:
+        """Restrict adding other fields through this schema."""
+
+        extra = Extra.forbid
+
+
+class MessagingConfigRequest(BaseModel):
+    """Messaging Config Request Schema"""
 
     name: str
     key: Optional[FidesOpsKey]
-    service_type: EmailServiceType
-    details: Union[
-        EmailServiceDetailsMailgun,
-    ]
+    service_type: MessagingServiceType
+    details: Optional[Union[MessagingServiceDetailsMailgun]]
```

Comment:
Do we need a union if there is only one option?

Reply:
nope, I'll remove